plugins/Jobs.cpp

```diff
@@ -160,32 +160,50 @@ bool BedrockPlugin_Jobs::peekCommand(SQLite& db, BedrockCommand& command) {
     } 
     
     // ----------------------------------------------------------------------
-    else if (SIEquals(request.methodLine, "CreateJob")) {
-        // If unique flag was passed and the job exist in the DB, then we can
-        // finish the command without escalating to master.
-        if (!request.test("unique")) {
-            // Not unique; need to process
-            return false;
+    else if (SIEquals(request.methodLine, "CreateJob") || SIEquals(request.methodLine, "CreateJobs")) {
+        vector<STable> jsonJobs;
+        if (SIEquals(request.methodLine, "CreateJob")) {
+            jsonJobs.push_back(request.nameValueMap);
+        } else {
+            list<string> multipleJobs;
+            multipleJobs = SParseJSONArray(request["jobs"]);
+            for (auto& job : multipleJobs) {
+                jsonJobs.push_back(SParseJSONObject(job));
+            }
         }
 
-        // Verify unique
-        SQResult result;
-        SINFO("Unique flag was passed, checking existing job with name " << request["name"]);
-        if (!db.read("SELECT jobID, data "
-                     "FROM jobs "
-                     "WHERE name=" + SQ(request["name"]) + ";",
-                     result)) {
-            throw "502 Select failed";
-        }
-        // If there's no job, or the existing job doesn't match the data we've been passed, escalate to master.
-        if (result.empty() || result[0][1] != request["data"]) {
-            return false;
-        }
+        for (auto& job : jsonJobs) {
+            // If unique flag was passed and the job exist in the DB, then we can
+            // finish the command without escalating to master.
+            if (!SContains(job ,"unique") || job["unique"] == "false") {
+                // Not unique; need to process
+                return false;
+            }
+
+            // Verify unique
+            SQResult result;
+            SINFO("Unique flag was passed, checking existing job with name " << job["name"]);
+            if (!db.read("SELECT jobID, data "
+                         "FROM jobs "
+                         "WHERE name=" + SQ(job["name"]) + ";",
+                         result)) {
+                throw "502 Select failed";
+            }
+            // If there's no job, escalate to master.
+            if (result.empty()){
+                return false;
```

Comment:
This is escalating to master based on checks for a single job, but we could check that the data passed is correct for all the jobs before escalating to master to prevent escalating a command that we know will fail